base/server/src/com/netscape/cms/realm/PKIRealm.java

```diff
@@ -132,10 +145,21 @@ public Principal authenticate(final X509Certificate[] certs) {
 
             return getPrincipal(username, authToken);
 
-        } catch (Exception e) {
+        } catch (EMissingCredential | EInvalidCredentials e) { // authentication failure
 
             logger.warn("Unable to authenticate cert chain: " + e.getMessage());
```

Comment:
(same comment)
Should we log Auth failure as `error`? Should we include stacktrace?

Reply:
See the other comment.